### PR TITLE
Make allocators' resolveInternalPointer take a const pointer as source

### PR DESCRIPTION
**Pull request by edi33416**

### CODE REVIEW

File: std/experimental/allocator/gc_allocator.d [hunk 1]
```diff
@@ -70,9 +70,10 @@ struct GCAllocator
     }
 
     /// Ditto
-    pure nothrow Ternary resolveInternalPointer(void* p, ref void[] result) shared
+    pure nothrow
```
Comment (andralex) [inline std/experimental/allocator/gc_allocator.d]: I think this entire function can be trusted, maybe in a later step

File: std/experimental/allocator/gc_allocator.d [hunk 2]
```diff
@@ -70,9 +70,10 @@ struct GCAllocator
...
+    Ternary resolveInternalPointer(const void* p, ref void[] result) shared
     {
-        auto r = GC.addrOf(p);
+        auto r = GC.addrOf(cast(void*)p);
```
Comment (CyberShadow) [inline std/experimental/allocator/gc_allocator.d]: Style check failed:
```
Enforce space after cast(...)
grep -nrE '[^&quot;]cast\([^)]*?\)[[:alnum:]]' $(find . -name '*.d') ; test $? -eq 1
./std/experimental/allocator/gc_allocator.d:76:        auto r = GC.addrOf(cast(void*)p);
```
Please fix/revert :)

**Comment by schveiguy** [comment]
> Please fix/revert :)

#5325

**Comment by schveiguy** [comment]
ping @wilzbach can dlang-bot not auto-merge when circleci complains?

**Comment by CyberShadow** [comment]
> ping @wilzbach can dlang-bot not auto-merge when circleci complains?

https://github.com/dlang-bots/dlang-bot/issues/69 (or we can set CircleCI as required).

**Comment by andralex** [comment]
Yah that'd be useful, thanks.

**Comment by wilzbach** [comment]
>  we can set CircleCI as required

Yep, that would be the easiest solution...

**Comment by CyberShadow** [comment]
@wilzbach Should we do it? Seems like we're in agreement.

**Comment by wilzbach** [comment]
> @wilzbach Should we do it? Seems like we're in agreement

Yes please! I have been trying to push for this since last summer :)

**Comment by CyberShadow** [comment]
> Yes please!

OK, I've set CircleCI as required for Phobos for now, we can expand to other repos if this works well.

> I have been trying to push for this since last summer :)

Oh, I knew there was a sentiment in favor of doing it, but I don't think I've received an explicit request to enable it in my inbox, otherwise I probably would have :)

**Comment by wilzbach** [comment]
> Oh, I knew there was a sentiment in favor of doing it, but I don't think I've received an explicit request to enable it in my inbox, otherwise I probably would have :)

Well, there were always a few problems with our CI, so with "pushing" I meant more wanting to make it happen. 
Thanks a lot!

**Comment by wilzbach** [comment]
I know that this is in `std.experimental`, but Allocator is treated as nearly "standard" by the community, so breaking changes do affect projects, e.g. https://github.com/rejectedsoftware/vibe.d/pull/1774
At least you should mention breaking changes in the changelog!

**Pull request merged**